src/connection_string.ts

```diff
@@ -70,97 +70,91 @@ function matchesParentDomain(srvAddress: string, parentDomain: string): boolean
  * @param uri - The connection string to parse
  * @param options - Optional user provided connection string options
  */
-export function resolveSRVRecord(options: MongoOptions, callback: Callback<HostAddress[]>): void {
+export async function resolveSRVRecord(options: MongoOptions): Promise<HostAddress[]> {
+  // We need to do this here to make sinon tests work :(
```

Comment:
this can't be done outside of `resolveSRVRecord`?

Reply:
switching to promises and fixed the sinon usages